lib/pure/decimal.nim

```diff
@@ -0,0 +1,1040 @@
+  ## 1E+3      (aka 1???.?????????...) has -3 places
+  ## 
+  ## Zero can also be given decimal places.
+  ## 
+  ## 0         (aka 0.??????????...) has 0 places
+  ## 0.00000   (aka 0.00000?????...) has 5 places
+  ## 
+  ## "0.00000" means the number is precisely zero to 5 decimals places.
+  ## 
+  ## Infinite and NaN have no decimal places and will return a zero (0).
+  let (decKind, exponent, upper) = determineKindExponentAndUpper(number)
+  case decKind:
+  of dkValued:
+    result = -exponent
+  of dkInfinite:
+    result = 0
+  of dkNaN:
+    result = 0
+
+proc `places=`*(number: var Decimal, newValue: int) {.inline.} =
+  ## Change a Decimal to the supplied number decimal places.
+  ##
+  ## The scale must be a value from −6144 to +6143.
+  ## 
+  ## A negative value means the significance is adjusted so that the
+  ## value is only accurate to the number of digits *before* the decimal place.
+  ## 
+  ## For example:
+  ##     var x = newDecimal("123.4")
+  ##     x.places = -1
+  ##     assert $x == "12E1" # essentially, 120 rounded to the nearest ten.
+  ##  
+  ## When NaN or Infinity is passed, the decimal is unchanged.
+  let (decKind, exponent, upper) = determineKindExponentAndUpper(number)
+  case decKind:
+  of dkValued:
+    if newValue == noValue:
+      return
+    let currentPlaces = -exponent
+    if currentPlaces != newValue:
+      let digits = generateDigits(number, upper)
+      var diff = (currentPlaces - newValue).int16
+      if diff > 0: # take away precision
+        let temp = digits.toTempSignificandArray()
+        let (newDigits, roundingEffect) = bankersRoundingToEven(temp, diff)
+        number = createDecimal(number.negative, newDigits.toTempSignificandArray, exponent + roundingEffect)
+      else:
+        let significance = digitCount(digits)
+        let newSig = significance - diff
+        if newSig > significandSize:
+          raise newException(
+            ValueError, 
+            "Too many decimal places ($1 places for $2). A 128-bit decimal can only hold 34 digits ($3 rose to $4)."
+              .format(newValue, $number, significance, newSig)
+            )
+        else:
+          let newDigits = shiftDecimalsLeftWithZero(digits, -diff)
+          number = createDecimal(number.negative, newDigits.toTempSignificandArray, exponent + diff)
+  of dkInfinite:
+    discard
+  of dkNaN:
+    discard
+
+
+#
+# ops
+#
+
+proc `~==`(left: Decimal, right: Decimal): bool =
+  ## Determines whether two numbers match in the context of their
+  ## least-common significance.
+  ## 
+  ## For example:
+  ## 
+  ##   assert (1.1'm  ~== 1.0'm ) == false
+  ##   assert (1.0'm  ~== 1.0'm ) == true
+  ##   assert (1.0'm  ~== 1.01'm) == true
+  ##   assert (1.01'm ~== 1.0'm ) == true
+  ##   assert (1.01'm ~== 1.00'm) == false
+  ## 
+  ## Use this with care with currency or financial work:
+  ## 
+  ##   assert (5'm    ~== 5.23'm) == true
+  ##   assert (5.00'm ~== 5.23'm) == false 
```

Comment:
```suggestion
  runnableExamples:
    assert (1.1'm  ~== 1.0'm ) == false
    assert (1.0'm  ~== 1.0'm ) == true
    assert (1.0'm  ~== 1.01'm) == true
    assert (1.01'm ~== 1.0'm ) == true
    assert (1.01'm ~== 1.00'm) == false
  ##
  ## Use this with care with currency or financial work:
  ##
  runnableExamples:
    assert (5'm    ~== 5.23'm) == true
    assert (5.00'm ~== 5.23'm) == false 
```
(Also, you can try to avoid trailing whitespace, for example by automatically removing it on save in your editor).

Reply:
fixed on next commit.